src/common/expression/Expression.h

```diff
@@ -223,6 +224,17 @@ class Expression {
 
 std::ostream& operator<<(std::ostream& os, Expression::Kind kind);
 
+struct PropertyTracker {
```

Comment:
This class should not be part of this file i think

Reply:
I will find a better place to put it into.